compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -6200,7 +6200,12 @@ void solveSelectTypeAndResolveType(BLangQueryExpr queryExpr, BLangExpression sel
             case TypeTags.NONE:
             default:
                 // contextually expected type not given (i.e var).
-                selectType = checkExpr(selectExp, env, type, data);
+                selectType = checkExprSilent(nodeCloner.cloneNode(selectExp), type, data);
```

Comment:
Regarding the comment at line 6202, as the default case is being used in the scenarios that the contextually expected type is not provided, don't we need to add additional cases for missing types in the switch statement? If not, can we remove the comment?